### PR TITLE
txn: Support bypass specified locks for reads

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

###  What have you changed?

This is a part of supporting large transaction.

When an transaction T1 meets another large transaction T2's lock, after invoking `check_txn_status` on T2, if T2 is not committed and `check_txn_status` enlarged T2's `min_commit_ts` to a value greater than `T1.start_ts`, then T1 can bypass T2's lock to read earlier value. This PR supports this functionality.

**This is a critical change. Please be very careful when reviewing this PR.**

###  What is the type of the changes?

- New feature (a change which adds functionality)

###  How is the PR tested?

- Unit test
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by MyonKeminta** [comment]
I want to run a benchmark to reveal whether there is significant performance regression caused by this PR but it seems the auto benchmark doesn't work...

**Comment by MyonKeminta** [comment]
It seems it doesn't harm performance much.

**Comment by MyonKeminta** [comment]
@youjiali1995 @sticnarf @coocood PTAL carefully.

File: src/storage/mvcc/mod.rs [hunk 1]
```diff
@@ -29,6 +33,63 @@ pub fn compose_ts(physical: u64, logical: u64) -> u64 {
...
+
+impl TsSet {
+    /// Create a `TsSet` from the given vec of timestamps. It will select the proper internal
+    /// collection type according to the size.
+    #[inline]
+    pub fn new(ts: Vec<u64>) -> Self {
+        if ts.is_empty() {
+            TsSet::Empty
+        } else if ts.len() <= TS_SET_USE_VEC_LIMIT {
+            // If there are too less elements in `ts`, use Vec directly instead of making a
```
Comment (youjiali1995) [inline src/storage/mvcc/mod.rs]:
```suggestion
            // If there are too few elements in `ts`, use Vec directly instead of making a Set.
```

Comment (MyonKeminta) [reply src/storage/mvcc/mod.rs]: 😂

**Comment by sticnarf** [comment]
> It seems it doesn't harm performance much.

sysbench is almost oltp. I'd like to try some AP scenarios...

**Comment by coocood** [comment]
LGTM

**Comment by sticnarf** [comment]
Actually I think the `Arc` is avoidable, but it should introduce some lifetimes in `ScannerConfig` and `PointGetter`. Probably not that necessary...

**Comment by MyonKeminta** [comment]
@sticnarf I think it's sometimes avoidable in the current structure, since the SnapshotScanner sometimes needs impl Send to move to another thread.

**Pull request merged**